SDK/AppCenterPush/Microsoft.AppCenter.Push.UWP/Push.cs

```diff
@@ -107,6 +129,7 @@ private void ApplyEnabledState(bool enabled)
             }
             else if (_channel != null)
             {
+                UserIdContext.UserIdUpdated -= OnUserIdUpdated;
```

Comment:
Reset LatestPushToken to null as well, otherwise when reenabling we won't resend it with a possibly new userId change that we missed while we were disabled.

Reply:
And a test for it ;)

Reply:
Thank you for bringing up this scenario. I will add a test for if the push service has been disabled, and then when user enables the service again, we can still listen for the changes. 

But isn't it actually we should not set `LatestPushToken` to null here since `OnUserIdUpdated` listens for user id changes only when the token is not null. If we set to null, we cannot reproduce the event after the push service is enabled since the user id has already changed.

Reply:
Correct I actually had last userId in mind and was confusing the 2 variables. Looks like only a test is needed then.